### PR TITLE
pysemgrep: use ATD classes in app/scans.py

### PR DESCRIPTION
**Pull request by aryx**
test plan:
precommit run -a mypy

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: cli/src/semgrep/app/scans.py [hunk 1]
```diff
@@ -297,14 +295,19 @@ def report_findings(
...
+        api_scans_findings = out.ApiScansFindings(
             # send a backup token in case the app is not available
-            "token": token,
-            "findings": findings,
-            "searched_paths": [str(t) for t in sorted(targets)],
+            token=token,
+            findings=findings,
+            searched_paths=[str(t) for t in sorted(targets)],
+            rule_ids=rule_ids,
+            cai_ids=cai_ids,
```
Comment (chmccreery) [inline cli/src/semgrep/app/scans.py:304]: @aryx can you explain why the cai_ids are still here?

Comment (aryx) [reply cli/src/semgrep/app/scans.py:304]: I didn't update yet the code to the latest semgrep-interfaces. I just realized that the current code was not using the classes defined in semgrep_output_v1.atd, so I first changed the code to use the classes, so then when I update semgrep-interfaces I'll get mypy error for the use of an undefined field.

**Comment by aryx** [comment]
after this PR, when I update semgrep-interfaces, I get:
```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

cli/src/semgrep/app/scans.py:298: error: Unexpected keyword argument "cai_ids" for "ApiScansFindings"  [call-arg]
Found 1 error in 1 file (checked 137 source files)

mypy (test files)........................................................Passed
```
which is nice

**Comment by aryx** [comment]
I'll make the PR that update to the latest semgrep-interfaces now (and remove cai_ids).

**Comment by chmccreery** [comment]
> I'll make the PR that update to the latest semgrep-interfaces now (and remove cai_ids).

Ah, perfect, thanks!

**Pull request merged**